### PR TITLE
Bubblegum parse

### PR DESCRIPTION
**Pull request by ngundotra**
Read NFT metadata into postgres

### CODE REVIEW

File: nft_api/src/ingest/bubblegum_indexer.rs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            // Retrieve nonce value from the LeafSchemaEvent emitted
+            let leaf_event_result = get_bubblegum_leaf_event(logs);
+            if leaf_event_result.is_err() {
+                println!("Could not find leaf event");
+                return Err(());
+            };
+            let leaf_event = leaf_event_result.unwrap();
+
+            let owner = pubkey_from_fb_table(keys, instruction.accounts[4] as usize);
+            let delegate = pubkey_from_fb_table(keys, instruction.accounts[4] as usize);
```
Comment (jarry-xiao) [inline nft_api/src/ingest/bubblegum_indexer.rs:85]: These are the same on mint right

File: programs/bubblegum/src/lib.rs [hunk 2]
```diff
@@ -265,6 +265,36 @@ pub struct Compress<'info> {
...
+    Decompress,
+}
+pub fn get_instruction_type(full_bytes: &Vec<u8>) -> InstructionName {
+    let disc: [u8; 8] = {
+        let mut disc = [0; 8];
+        disc.copy_from_slice(&full_bytes[..8]);
+        disc
+    };
+    match disc {
+        [51, 57, 225, 47, 182, 146, 137, 166] => InstructionName::Mint,
```
Comment (jarry-xiao) [inline programs/bubblegum/src/lib.rs:288]: I wonder if there's a better way to do this. There's probably a `discriminator` Trait that's implemented for each of the instructions

**Comment by jarry-xiao** [comment]
Looks good to me!

File: programs/bubblegum/src/lib.rs (hunk 2 above)
Comment (ngundotra) [reply programs/bubblegum/src/lib.rs:288]: fair point, will check this one more time before committing

File: nft_api/src/ingest/bubblegum_indexer.rs (hunk 1 above)
Comment (ngundotra) [reply nft_api/src/ingest/bubblegum_indexer.rs:85]: Yes they are

Comment (ngundotra) [reply nft_api/src/ingest/bubblegum_indexer.rs:85]: Actually, they are not

File: programs/bubblegum/src/lib.rs (hunk 2 above)
Comment (ngundotra) [reply programs/bubblegum/src/lib.rs:288]: No there isn't a discriminator trait that does this

**Pull request merged**